### PR TITLE
scripts: remove boards_legacy sub-folder from list_boards.py

### LINKED ISSUE #69785: hwmv2: Out of tree hwmv1 boards/socs do not work
**Describe the bug**
Seemingly getting a second zephyr checkout on 3.6 and commenting the deprecation of a board then prevents the original board from being found. Indeed it seems that `scripts/list_boards.py` just returns nothing at all when using the SOC and BOARD roots of the zephyr 3.6 directory.

**To Reproduce**
Comment out a board deprecating and try building for it with BOARD_ROOT and SOC_ROOT set to the zephyr 3.6 directory. As a bonus just run the list_boards script, e.g. `/tmp/aa/zephyr/scripts/list_boards.py --board-root=/tmp/bb/zephyr/ --arch-root=/tmp/aa/zephyr --soc-root=/tmp/bb/zephyr/` gives no output at all. Running the 3.6 script with the same parameters minus SOC root gives a full output `/tmp/bb/zephyr/scripts/list_boards.py --board-root=/tmp/bb/zephyr/ --arch-root=/tmp/aa/zephyr`:
```
...
nios2:
  altera_max10
  qemu_nios2
posix:
  native_posix
  native_posix_64
  native_sim
  native_sim_64
  nrf52_bsim
  nrf5340bsim_nrf5340_cpuapp
  nrf5340bsim_nrf5340_cpunet
riscv:
  adp_xc7k_ae350
  beaglev_fire
  esp32c3_devkitm
  esp32c3_luatos_core
  esp32c3_luatos_core_usb
...
```

**Expected behavior**
Boards to be detected and work

**Impact**
Showstopper, release blocker

**Environment (please complete the following information):**
 - Commit SHA or Version used: 72a17b6d887ef20b3cc08a25d8ea1d622792d57a

### PR DESCRIPTION
**Pull request by tejlmand**
Fixes: #69785

The boards_legacy sub-folder was temporarily introduce in collab-hwm branch during porting to HWMv2.

This should have been removed before merging collab-hwm to main as it prevent looking up boards in oot roots.

Removing the temporary sub-folder for HWMv2.

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

File: cmake/modules/hwm_v2.cmake [hunk 1]
```diff
@@ -22,6 +22,8 @@ if(NOT HWMv2)
   return()
 endif()
 
+set_ifndef(KCONFIG_BINARY_DIR ${CMAKE_BINARY_DIR}/Kconfig)
```
Comment (golowanow) [inline cmake/modules/hwm_v2.cmake:25]: I wonder how is this change related to what is put into the commit description ?

**Review by MaureenHelm — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]
@tejlmand could you take a look at @golowanow comment? that line does indeed looks off, at least it does to me as well, just laying on the safe side making sure it's intentional

**Comment by MaureenHelm** [comment]
> @tejlmand could you take a look at @golowanow comment? that line does indeed looks off, at least it does to me as well, just laying on the safe side making sure it's intentional

@tejlmand ping. I have some OOT SoC/boards waiting for this fix.

File: scripts/list_boards.py [hunk 2]
```diff
@@ -150,10 +150,10 @@ def find_arch2board_set_in(root, arches, board_dir):
     boards = root / 'boards'
 
     for arch in arches:
-        if not (boards / "boards_legacy" / arch).is_dir():
+        if not (boards / arch).is_dir():
             continue
 
-        for maybe_board in (boards / "boards_legacy" / arch).iterdir():
+        for maybe_board in (boards / arch).iterdir():
```
Comment (57300) [inline scripts/list_boards.py:156]: With this change by itself, all Arm boards will appear as both v1 and v2:
```
> west boards | grep mps2
mps2_an385
mps2_an521_cpu0
mps2_an521_cpu0_ns
mps2_an521_cpu1
mps2
```

**Review by 57300 — CHANGES_REQUESTED** [review]
Note: OOT boards won't work unless their OOT SoCs are moved to `<SOC_ROOT>/soc/soc_legacy`.
However, the required Kconfig changes seem non-trivial, so perhaps this shouldn't be a blocker.

File: cmake/modules/hwm_v2.cmake (hunk 1 above)
Comment (57300) [reply cmake/modules/hwm_v2.cmake:25]: Afaict, the initial value of `KCONFIG_BINARY_DIR` is usually set by `zephyr_module.cmake`. However, that file is also loaded in unit test mode, so I'm not sure why `hwm_v2.cmake` couldn't be loaded after it, instead of adding this line.

File: scripts/list_boards.py (hunk 2 above)
Comment (tejlmand) [reply scripts/list_boards.py:162]: Fixed.

**Comment by tejlmand** [comment]
> What about the `soc_legacy` folder?

Fixed

**Review by nordicjm — DISMISSED** [review]

File: cmake/modules/hwm_v2.cmake (hunk 1 above)
Comment (tejlmand) [reply cmake/modules/hwm_v2.cmake:25]: > so I'm not sure why `hwm_v2.cmake` couldn't be loaded after it, instead of adding this line.

Good question. I can't see why https://github.com/zephyrproject-rtos/zephyr/pull/64401 added the lines at the current locations instead of following the same structure used in `zephyr_default.cmake`.

Comment (tejlmand) [reply cmake/modules/hwm_v2.cmake]: Reordered `include`s to follow same order as zephyr_default.cmake.

**Review by nordicjm — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]
